src/routes/event/details.tsx

```diff
@@ -0,0 +1,96 @@
+import { useLoaderData, useNavigate } from "react-router-dom";
+import { database, auth } from "../../firebase";
+import { get, ref } from "firebase/database";
+import { Button, Container, Stack, TextField, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+
+export async function loader({ params }) {
+  return params.eventId;
+}
+
+type EventDetailsProps = {
+  eventCategory: string;
+  eventDescription: string;
+  eventEndDate: string;
+  eventLocation: string;
+  eventOrganizer: string;
+  eventStartDate: string;
+};
+
+type EventMetadataProps = {
+  eventName: string;
+};
+
+export default function EventDetails() {
+  const eventId = useLoaderData();
+  const [eventDetails, setEventDetails] = useState<EventDetailsProps>();
+  const [eventMetadata, setEventMetadata] = useState<EventMetadataProps>({
+    eventName: "",
+  });
+  const navigate = useNavigate();
+  useEffect(() => {
+    get(ref(database, `event-details/${eventId}`)).then((snapshot) => {
+      if (snapshot.exists()) {
+        setEventDetails(snapshot.val());
+      } else {
+        console.log("No data available");
+      }
+    });
+  }, []);
+  useEffect(() => {
+    get(ref(database, `event-names/${eventId}`)).then((snapshot) => {
+      if (snapshot.exists()) {
+        setEventMetadata(snapshot.val());
+      } else {
+        console.log("No data available");
+      }
+    });
+  }, []);
+  return (
+    <Container component="main" maxWidth="xs" disableGutters>
+      <Typography variant="h2" align="center" gutterBottom color={"cyan"}>
+        Event Details
+      </Typography>
+      <Stack component="form" spacing={2}>
+        <TextField label="Event name" defaultValue={eventMetadata} />
+        <TextField
+          label="Event description"
+          defaultValue={eventDetails?.eventDescription}
+        />
+        <TextField
+          label="Event start date"
+          defaultValue={eventDetails?.eventStartDate}
+        />
+        <TextField
+          label="Event end date"
+          defaultValue={eventDetails?.eventEndDate}
+        />
+        <TextField
+          label="Event location"
+          defaultValue={eventDetails?.eventLocation}
+        />
+        <TextField
+          label="Event category"
+          defaultValue={eventDetails?.eventCategory}
+        />
+        <Stack direction="row" spacing={2}>
+          {auth.currentUser &&
+            auth.currentUser.uid === eventDetails?.eventOrganizer && (
+              <Button variant="contained" color="primary">
+                Edit
+              </Button>
+            )}
+          <Button
+            variant="contained"
+            color="primary"
+            onClick={() => {
+              navigate("/event/list");
```

Comment:
you have this here for the 3rd time, so if you have time, you can check the above comment on this.